contrib/endpoints/src/api_manager/request_handler.cc

```diff
@@ -79,6 +96,26 @@ void RequestHandler::AttemptIntermediateReport() {
 // Sends a report.
 void RequestHandler::Report(std::unique_ptr<Response> response,
                             std::function<void(void)> continuation) {
+  if (context_) {
```

Comment:
For report, the logic should be:
 if Check() has registered a callback,  it is not called yet,   just abort the report.  Otherwise, just continue the normal report.